File: SRP Redux/Exercise/MarsRoverKata/Commands/TurnRightCommand.cs
namespace MarsRoverKata.Commands
{
    public class TurnRightCommand : ICommand
    {
        private Rover Rover { get; set; }

        public TurnRightCommand(Rover rover)
        {
            Rover = rover;
        }

        public bool Execute()
        {
            return Rover.TurnRight();
        }
    }
}

File: SRP/Java_Final/MarsRoverKata/src/main/java/MarsRoverKata/Commands/TurnRightCommand.java
package MarsRoverKata.Commands;

import MarsRoverKata.Rover;

public class TurnRightCommand implements ICommand {
    private Rover rover;

    public Rover getMovable() {
        return rover;
    }

    public void setRover(Rover movable) {
        this.rover = movable;
    }

    public TurnRightCommand(Rover movable) {
        this.rover = movable;
    }

    public boolean execute() {
        return rover.turnRight();
    }

}
